admin/server/deployment.go

```diff
@@ -830,6 +830,43 @@ func (s *Server) GetIFrame(ctx context.Context, req *adminv1.GetIFrameRequest) (
 	}, nil
 }
 
+func (s *Server) GetDeploymentConfig(ctx context.Context, req *adminv1.GetDeploymentConfigRequest) (*adminv1.GetDeploymentConfigResponse, error) {
+	claims := auth.GetClaims(ctx)
+	// TODO: is it sufficient to check only OwnerTypeDeployment here or other permission checks are needed?
+	if claims.OwnerType() != auth.OwnerTypeDeployment {
+		return nil, status.Error(codes.PermissionDenied, "only deployments can get their config")
+	}
```

Comment:
It could be, but since there's a `GetDeploymentConfigRequest.deployment_id` setting, that should probably be removed or supported as well (for any user with manage prod/dev permissions). 

Maybe use this snippet from the `Provision` RPC?
```
if req.DeploymentId == "" {
		if claims.OwnerType() == auth.OwnerTypeDeployment {
			req.DeploymentId = claims.OwnerID()
		} else {
			return nil, status.Error(codes.InvalidArgument, "missing deployment_id")
		}
	}
```